suro-elasticsearch/src/main/java/com/netflix/suro/sink/elasticsearch/ElasticSearchSink.java

```diff
@@ -264,90 +160,199 @@ public String getStat() {
     @Override
     protected void beforePolling() throws IOException {}
 
-    private IndexRequest createIndexRequest(Message m) {
+    private void createClient() {
+        if (ribbonEtc.containsKey("eureka")) {
+            ribbonEtc.setProperty(
+                clientName + ".ribbon.AppName", clientName);
+            ribbonEtc.setProperty(
+                clientName + ".ribbon.NIWSServerListClassName",
+                "com.netflix.niws.loadbalancer.DiscoveryEnabledNIWSServerList");
+            ribbonEtc.setProperty(
+                clientName + ".ribbon.ServerListRefreshInterval",
+                "60000");
+            String[] host_port = addressList.get(0).split(":");
+            ribbonEtc.setProperty(
+                clientName + ".ribbon.DeploymentContextBasedVipAddresses",
+                host_port[0]);
+            ribbonEtc.setProperty(
+                clientName + ".ribbon.Port",
+                host_port[1]);
```

Comment:
Just in case, the application is not providing port number to register itself in eureka server, I need to explicitly set up the port property.